tests/x11/rstudio_server.pm

```diff
@@ -62,13 +62,19 @@ sub run() {
     # log out at last and close firefox
     assert_and_click("rstudio_server-sign-out");
     assert_screen("rstudio_server-login-username");
+
     send_key('alt-f4');
 
-    # optionally click away the close all tabs window
-    assert_screen([qw(rstudio_server-firefox_quit-and-close-tabs generic-desktop)]);
+    check_screen([qw(rstudio_server-firefox_quit-and-close-tabs generic-desktop)], timeout => 60);
     if (match_has_tag('rstudio_server-firefox_quit-and-close-tabs')) {
+        # optionally click away the close all tabs window => FF should close
         click_lastmatch();
         assert_screen('generic-desktop');
+    } elsif (!match_has_tag('generic-desktop')) {
+        # firefox is stuck and just greys out but does not display the "close
+        # all tabs" dialog => kill it
+        script_run('pkill -9 firefox');
+        assert_screen('generic-desktop');
```

Comment:
this seems it belong outside the conditional block. and can be removed from L72 too

Reply:
Indeed, thanks!